azurerm/internal/services/netapp/netapp_pool_resource.go

```diff
@@ -216,15 +224,6 @@ func netappPoolDeleteStateRefreshFunc(ctx context.Context, client *netapp.PoolsC
 			}
 		}
 
-		// The resource NetApp Pool depends on the resource NetApp Account.
-		// Although the delete API returns 404 which means the NetApp Pool resource has been deleted.
-		// Then it tries to immediately delete NetApp Account but it still throws error `Can not delete resource before nested resources are deleted.`
-		// In this case we're going to try triggering the Deletion again, in-case it didn't work prior to this attempt.
-		// For more details, see related Bug: https://github.com/Azure/azure-sdk-for-go/issues/6374
-		if _, err := client.Delete(ctx, resourceGroupName, accountName, name); err != nil {
-			log.Printf("Error reissuing NetApp Pool %q delete request (Resource Group %q): %+v", name, resourceGroupName, err)
-		}
```

Comment:
is this API bug resolved? this continually deletes since the API only worked 50% of the time previously

Reply:
After tried, the bug still hasn't been fixed. Once this PR got merged, I assume pass rate can be improved. After tested several times, they all passed.